### PR TITLE
cypress setup and guest Sign in e2e test case

### PR DESCRIPTION
**Pull request by subhashkhileri**
cypress setup and guest Sign in e2e test case

[Issue : https://github.com/janus-idp/showcase-e2e/issues/46](https://github.com/janus-idp/showcase-e2e/issues/46)

### CODE REVIEW

File: e2e-test/cypress/support/e2e.ts [hunk 1]
```diff
@@ -31,6 +21,10 @@ import {
...
  */
-Cypress.Commands.add("login", () => {
-  cy.visit("/");
+
+// Handling errors from application
+// eslint-disable-next-line no-unused-vars,@typescript-eslint/no-unused-vars
+Cypress.on('uncaught:exception', (err) => {
+  return false;
 });
+
```
Comment (josephca) [inline e2e-test/cypress/support/e2e.ts:30]: Minor caveat: suggest adding an extra line.

File: e2e-test/cypress/support/pageObjects/global-obj.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export const waitsObjs = {
+    MuiLinearProgress: 'div[class*="MuiLinearProgress-root"]',
+    MuiCircularProgress: '[class*="MuiCircularProgress-root"]',
+};
+
+export const UIhelperPO = {
+    buttonLabel: 'span[class^="MuiButton-label"]',
+    MuiBoxLabel:'div[class*="MuiBox-root"] label',
+    MuiTableRow: 'tr[class*="MuiTableRow-root"]',
+}
```
Comment (josephca) [inline e2e-test/cypress/support/pageObjects/global-obj.ts:10]: Minor caveat: suggest adding an extra line.

File: e2e-test/cypress/support/pageObjects/page-obj.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const HomePagePO = {
+    searchBar: 'input[aria-label="Search"]',
+    MuiAccordion: 'div[class*="MuiAccordion-root-"]',
+}
+
+export const SettingsPagePO = {
+    userSettingsMenu: 'button[data-testid="user-settings-menu"]',
+    signOut: 'li[data-testid="sign-out"]',
+}
```
Comment (josephca) [inline e2e-test/cypress/support/pageObjects/page-obj.ts:9]: Minor caveat: suggest adding an extra line.

File: e2e-test/cypress/support/testData/gpts.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const gpts = [
+    'Add ArgoCD to an existing project', 'Ansible Job Template', 'Create a .NET Frontend application in Azure DevOps with a CI pipeline', 'Create a Go Backend application with a CI pipeline',
+    'Ansible Job', 'Create a Node.js Backend application with a CI pipeline', 'Create a Python application in GitLab with a CI pipeline', 'Create a Quarkus Backend application with a CI pipeline',
+    'Register existing component to Software Catalog', 'Create a Spring Boot Backend application with a CI pipeline', 'Create a techdocs sample', 'Create a tekton CI Pipeline'
+]
```
Comment (josephca) [inline e2e-test/cypress/support/testData/gpts.ts:5]: Minor caveat: suggest adding an extra line.

File: e2e-test/cypress/utils/Common.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        cy.get(SettingsPagePO.signOut).click();
+        UIhelper.verifyHeading('Select a sign-in method');
+    }
+
+    static waitForLoad(timeout = 120000) {
+        for (const item of Object.values(waitsObjs)) {
+            cy.get(item, { timeout }).should('not.exist');
+        }
+    }
+}
```
Comment (josephca) [inline e2e-test/cypress/utils/Common.ts:25]: Minor caveat: suggest adding an extra line.

**Review by josephca — APPROVED** [review]

**Review by rnapoles-rh — APPROVED** [review]

**Comment by josephca** [comment]
@subhashkhileri rebase by grabbing https://github.com/janus-idp/showcase-e2e/pull/41 and re-run the test.

**Pull request merged**